### ISSUE TITLE
[feature request] SidecarSet support pre-download image

### ISSUE THREAD

**Issue by zmberg** [openkruise/kruise#1221]
**What would you like to be added**:

**Why is this needed**:

**Comment by zmberg** [comment]
@Hanmengnan This is GLCC project, are you student, and If you are, i suggest you can apply for this issue in glcc.

**Comment by Hanmengnan** [comment]
Yes, I am a student and I have also noticed that this is a GLCC project. However, I did not find a registration entry for this project under the Openkuise section of the GLCC official website. Can I still register for this project now? Can you give me some guidance or more other information? Thank you very much!

**Comment by zmberg** [comment]
> Yes, I am a student and I have also noticed that this is a GLCC project. However, I did not find a registration entry for this project under the Openkuise section of the GLCC official website. Can I still register for this project now? Can you give me some guidance or more other information? Thank you very much!

Yes, we can communication in github.

**Comment by Jayant-kernel** [comment]
@zmberg 
 I’d like to work on this issue.

From what I understand, this is about adding image pre-download support for SidecarSet, similar to the existing CloneSet flow, to make sidecar upgrades smoother.

Is this still something you’d like to have? If yes, I’d appreciate any guidance on the preferred direction before I start.
